Src/DependencyCollector/Shared/Implementation/Operation/CacheProvider.cs

```diff
@@ -3,392 +3,34 @@
     using System;
     using System.Collections.Generic;
     using System.Globalization;
-    using System.Threading;
+    using System.Runtime.Caching;
+    using System.Threading;        
     using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
 
     /// <summary>
-    /// The implementation of cache provider for Windows Phone and Windows Store as MemoryCache is not available there.
+    /// The wrapper for MemoryCache.
     /// </summary>
     /// <typeparam name="TValue">Type of items to store in the cache.</typeparam>
-    internal sealed class CacheProvider<TValue> : ICacheProvider<TValue>
+    internal class CacheProvider<TValue> : ICacheProvider<TValue>
```

Comment:
Why keep the class at all?
Unless it is used heavily in multiple places I think we should just switch the code that uses this class to use MemoryCache directly instead and delete the rest of it.